packages/sqs/lib/sqs/AbstractSqsConsumer.ts

```diff
@@ -54,8 +55,9 @@ export type SQSConsumerOptions<
   SQSCreationConfig,
   SQSQueueLocatorType
 > & {
-  consumerOverrides?: Partial<ConsumerOptions>
+  consumerOverrides?: Pick<ConsumerOptions, 'terminateVisibilityTimeout' | 'heartbeatInterval'>
```

Comment:
`Partial<ConsumerOptions>` was allowing to override `handler`, `queueUrl` or `sqsClient` which is not good from my point of view.

I think that it is a good idea to not just expose everything without knowing how it works, so just exposing what we are using but please feel free to contradict me if you disagree

Reply:
yeah, fair point.
are there any other potentially useful options?

Reply:
Not sure if there are other useful options, do you think that makes sense to spend time researching other options now? 

The first version I wrote was using `Omit` -> https://github.com/kibertoad/message-queue-toolkit/pull/124/commits/e1bca7679e8ddcc0ddcadf6eba9011b30a19b853

but after giving it a thought I changed it because I don't know how other parameters behave and that can overlap with other things (eg: `visibilityTimeout`). My reasoning for choosing `Pick` is that I don't want to spend time now investigating all possibilities, and adding them later if needed is a minor release, but removing an overlap option is a major one.

But if you still think that Omit is a better option I will change it, I mean I see the benefits of exposing all possibilities to not have to work on a new version if the parameter is already there

Reply:
I would prefer to provide users an escape hatch for cases when SQS/SNS support something that we currently don't, and not force them to rely on us to expose something for them. So I would only hide the options that we rely upon, that might break if someone overrides them on top of our own dynamic setup.

I'm not even sure about hiding `visibilityTimeout`. Isn't it a perfectly legit use-case for someone to choose not to use the refresh with the heartbeat, but just to have some explicitly set static timeout?

Reply:
`visibilityTimeout` on consumer doesn't update the queue `VisibilityTimeout`, so I decided to hide it because both values should be in sync and is better to just pick the queue value to have a less confusing API, but please let me know if there is something I missed.

> I would prefer to provide users an escape hatch for cases when SQS/SNS support something that we currently don't, and not force them to rely on us to expose something for them.

I will change it to use Omit then, thanks for sharing Igor!

Reply:
Fair point, let's hide it from consumerOverrides and expect users to use our own option for this